examples/src/main/scala/com/example/http4s/ssl/SslExampleWithRedirect.scala

```diff
@@ -0,0 +1,48 @@
+package com.example.http4s
+package ssl
+
+import java.nio.file.Paths
+
+import fs2._
+import org.http4s.server._
+import org.http4s.dsl._
+import org.http4s.{HttpService, Uri}
+import org.http4s.headers.Host
+import org.http4s.server.SSLKeyStoreSupport.StoreInfo
+import org.http4s.server.{ SSLKeyStoreSupport, ServerBuilder }
+import org.http4s.util.StreamApp
+
+trait SslExampleWithRedirect extends StreamApp {
+  val securePort = 8443
+
+  implicit val strategy: Strategy = Strategy.fromFixedDaemonPool(2)
+
+  // TODO: Reference server.jks from something other than one child down.
+  val keypath = Paths.get("../server.jks").toAbsolutePath().toString()
+
+  def builder: ServerBuilder with SSLKeyStoreSupport
+
+  val redirectService = HttpService {
+    case request =>
+      request.headers.get(Host) match {
+        case Some(Host(host, _)) =>
```

Comment:
This won't work with a request that passes an absolute URI.  Maybe `baseUri = Uri.copy(scheme = "https".ci, port = securePort)`?